claim/services.py

```diff
@@ -626,15 +626,12 @@ def with_relative_prices(claim):
 
 def set_claims_status(uuids, field, status, audit_data=None, user=None):
     errors = []
-    for claim_uuid in uuids:
-        claim = Claim.objects \
-            .filter(uuid=claim_uuid,
-                    validity_to__isnull=True) \
-            .first()
-        if claim is None:
-            errors += [{'message': _(
-                "claim.validation.id_does_not_exist") % {'id': claim_uuid}}]
-            continue
+    claims = Claim.objects \
+            .filter(uuid__in=uuids,
+                    *filter_validity()) 
+    remaining_uuid = list(map(str.upper,uuids))
+    for claim in claims:
+        remaining_uuid.remove(claim.uuid.upper())        
         try:
             claim.save_history()
```

Comment:
You may have a look at bulk_create if I am not mistaken, it should be a single db call.

Reply:
So instead of saving claims one by one you save them all at once.

Reply:
indeed but we wont have the save history, tbd later